lib/index.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const autobind = require('auto-bind');
+const path = require('path');
+
+const defaultOptions = { ttl: 0 };
+
+class CacheManager {
+  constructor(options) {
+    options = { ...defaultOptions, ...options };
+    this.provider = CacheManager.createProvider(options);
+    autobind(this);
+  }
+
+  set(key, value, ttl) {
+    return this.provider.set(key, value, ttl);
+  }
+
+  get(key) {
+    return this.provider.get(key);
+  }
+
+  has(key) {
+    return this.provider.has(key);
+  }
+
+  getKeys(pattern = '*') {
+    return this.provider.getKeys(pattern);
+  }
+
+  delete(key) {
+    return this.provider.delete(key);
+  }
+
+  static createProvider({ store = 'memory', ...options }) {
```

Comment:
Does it make sense to keep defaults in one place by moving this one to the `defaultOptions`?